From 31754c9266763b123bdcc1613916b31fe05549b3 Mon Sep 17 00:00:00 2001
From: Yi Kong <yikong@google.com>
Date: Wed, 4 Nov 2020 01:44:15 +0800
Subject: [PATCH 8/8] Always generate profiling debug info

The additional debug info is used for sampling PGO. This does not affect
the size of final stripped binaries.

Also removed `-gmlt` flag in pgo.go, since it's actually a lower level
of debug option than the default (-g).

Test: build, verify final binaries are identical size
Bug: 79161490
Change-Id: Ie4cf4998695f78a50a7ac7a7a0853c625cbd8880
---
 cc/config/global.go | 1 +
 cc/pgo.go           | 4 ----
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/cc/config/global.go b/cc/config/global.go
index 473c8067d..6e162043e 100644
--- a/cc/config/global.go
+++ b/cc/config/global.go
@@ -45,6 +45,7 @@ var (
 
 		"-O2",
 		"-g",
+		"-fdebug-info-for-profiling",
 
 		"-fno-strict-aliasing",
 
diff --git a/cc/pgo.go b/cc/pgo.go
index 6bf0ad0aa..02aa77119 100644
--- a/cc/pgo.go
+++ b/cc/pgo.go
@@ -41,7 +41,6 @@ var (
 var pgoProfileProjectsConfigKey = android.NewOnceKey("PgoProfileProjects")
 
 const profileInstrumentFlag = "-fprofile-generate=/data/local/tmp"
-const profileSamplingFlag = "-gmlt -fdebug-info-for-profiling"
 const profileUseInstrumentFormat = "-fprofile-use=%s"
 const profileUseSamplingFormat = "-fprofile-sample-accurate -fprofile-sample-use=%s"
 
@@ -100,9 +99,6 @@ func (props *PgoProperties) addInstrumentationProfileGatherFlags(ctx ModuleConte
 }
 func (props *PgoProperties) addSamplingProfileGatherFlags(ctx ModuleContext, flags Flags) Flags {
 	flags.Local.CFlags = append(flags.Local.CFlags, props.Pgo.Cflags...)
-
-	flags.Local.CFlags = append(flags.Local.CFlags, profileSamplingFlag)
-	flags.Local.LdFlags = append(flags.Local.LdFlags, profileSamplingFlag)
 	return flags
 }
 
-- 
2.31.1

